test/rules/avoid_relative_lib_imports.dart

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2022, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:test_reflective_loader/test_reflective_loader.dart';
+
+import '../rule_test_support.dart';
+
+main() {
+  defineReflectiveSuite(() {
+    defineReflectiveTests(AvoidRelativeLibImportsTest);
+  });
+}
+
+@reflectiveTest
+class AvoidRelativeLibImportsTest extends LintRuleTest {
+  @override
+  String get lintRule => 'avoid_relative_lib_imports';
+
+  test_packageImport() async {
+    newFile('$testPackageRootPath/lib/other.dart');
+    await assertNoDiagnostics(
+      r'''
+// ignore: unused_import
+import 'package:test/other.dart';
+''',
+    );
+  }
+
+  test_absolutePath() async {
+    var filePath = '$testPackageRootPath/lib/other.dart';
+    newFile(filePath);
+    await assertDiagnostics(
+      '''
+// ignore: unused_import
+import '$filePath';
+''',
+      [
+        lint('avoid_relative_lib_imports', 32, filePath.length + 2),
+        // TODO(dantup): Why???
+        error(CompileTimeErrorCode.URI_DOES_NOT_EXIST, 32, filePath.length + 2),
+      ],
+    );
+  }
+
+  test_relativePath_explicitLib() async {
+    var filePath = '$testPackageRootPath/lib/other.dart';
+    newFile(filePath);
+    await assertDiagnostics(
+      '''
+// ignore: unused_import
+import '../lib/other.dart';
+''',
+      [
+        lint('avoid_relative_lib_imports', 32, 19),
+        // TODO(dantup): Why???
```

Comment:
This is another odd case. I don't remember the justification for it, but I believe that analyzer rejects paths inside `lib` that escape `lib`, even if they re-enter `lib`. @scheglov would remember.

Reply:
We don't do this intentionally, but using `Uri.resolveUri()` does following:
1. When we resolve `package:foo/../lib/bar.dart`, we lose the package name, and get `package:./lib/bar.dart`, which cannot be resolved.
2. When we resolve `../lib/bar.dart` in `package:foo/foo.dart`, we get `package:foo/lib/bar.dart`, which might exist (if you have `lib` in `lib`), but usually is not.

Reply:
Ah, that explains this one. What should I do here? Should I add an `// ignore` for it, and a comment along those lines?

Any ideas why the same occurs for `test_absolutePath` above which is a full path (`/home/test/lib/other.dart`)?